src/pyjion/intrins.cpp

```diff
@@ -2120,22 +2119,50 @@ PyObject* MethCallN(PyObject* self, PyMethodLocation* method_info, PyObject* arg
             return nullptr;
         }
         auto obj =  method_info->object;
-        auto args_tuple = PyTuple_New(PyTuple_Size(args) + 1);
-        PyTuple_SET_ITEM(args_tuple, 0, obj);
-        Py_INCREF(obj);
-        for (int i = 0 ; i < PyTuple_Size(args) ; i ++){
-            PyTuple_SET_ITEM(args_tuple, i+1, PyTuple_GET_ITEM(args, i));
-            Py_INCREF(PyTuple_GET_ITEM(args, i));
-        }
+        if (PyCFunction_Check(target)) {
```

Comment:
@tonybaloney 
Is this what you meant here: https://github.com/tonybaloney/Pyjion/issues/24#issuecomment-753565826?
It looks like this checks whether it's a C function so I'm not able to trigger it from the unit tests I added. Is this the way it's meant to work?

Reply:
Yes, that's correct. I'm going to write some more c++ tests for these methods shortly (also to verify the reference counter impact, which I think is causing some bugs).

Reply:
Ok, I'll revert my Python tests.